src/OrchardCore.Modules/OrchardCore.Https/Views/HttpsSettings.Edit.cshtml

```diff
@@ -0,0 +1,27 @@
+@model HttpsSettingsViewModel
+
+<fieldset class="form-group" asp-validation-class-for="RequireHttps">
+    <div class="form-check">
+        <label class="form-check-label">
+            <input asp-for="RequireHttps" class="form-check-input" asp-is-disabled="@(!Model.IsHttpsRequest)" />
```

Comment:
Why is it disabled when the current request is https ?

Reply:
I shot myself in the foot and clicked enable require https over http when https settings were incorrect. This is only disabled if the current request is NOT https.

Reply:
So this way you can only enable it when the settings are correctly configured and working, meaning you are on HTTPS already? That's clever.

Reply:
Exactly